File: backend/app/models/analytics.py

from typing import List
from pydantic import BaseModel

class OrderStatusPercentage(BaseModel):
    status: str
    percentage: int


class MaterialTypePercentage(BaseModel):
    material_type: str
    percentage: int


class MonthlyOrder(BaseModel):
    month: str
    count: int


class MonthlyRevenue(BaseModel):
    month: str
    revenue: float


class DashboardAnalytics(BaseModel):
    total_users: int
    total_available_materials: int
    total_pending_orders: int
    total_today_appointments: int

    order_status_percentages: List[OrderStatusPercentage]
    material_type_percentages: List[MaterialTypePercentage]
    monthly_orders: List[MonthlyOrder]
    monthly_revenue: List[MonthlyRevenue]

File: frontend1/lib/types/analytics.ts
export interface DashboardAnalytics {
  total_users: number;
  total_available_materials: number;
  total_pending_orders: number;
  total_today_appointments: number;

  order_status_percentages: OrderStatusPercentage[];
  material_type_percentages: MaterialTypePercentage[];
  monthly_orders: MonthlyOrder[];
  monthly_revenue: MonthlyRevenue[];
}

export interface OrderStatusPercentage {
  status: string;
  percentage: number;
}

export interface MaterialTypePercentage {
  material_type: string;
  percentage: number;
}

export interface MonthlyOrder {
  month: string;  // e.g. "Jan", "Feb"
  count: number;
}

export interface MonthlyRevenue {
  month: string;  // e.g. "Jan", "Feb"
  revenue: number;
}
